include/oneapi/dpl/internal/scan_by_segment_impl.h

```diff
@@ -96,7 +96,8 @@ class __seg_scan_wg_kernel;
 template <bool __is_inclusive, typename... Name>
 class __seg_scan_prefix_kernel;
 
-template <bool __is_inclusive>
+// TODO required to move into oneapi::dpl::__par_backend_hetero namesace
```

Comment:
I think this should probably have a function that is another layer above this struct,  `oneapi::dpl::__par_backend_hetero::__parallel_scan_by_segment`, which then pulls the Name and queue out of the `ExecutionPolicy` then uses this struct.  With that in mind, the namespace of the struct itself is less important if it is not the entry point to the sycl device backend.

Reply:
@danhoeflinger I don't understood exactly what you mean here.
But usually the structures like that has placed in the `oneapi::dpl::__par_backend_hetero` namespace.
So in my opinion this structure now is placed in the wrong namespace.

Reply:
I described it in more detail in this comment: https://github.com/uxlfoundation/oneDPL/pull/2198#discussion_r2050818264 
In any case, lets make those adjustments in a separate PR, then rebase.

Reply:
#2201

Reply:
TODO has been removed.
The PR https://github.com/uxlfoundation/oneDPL/pull/2201 has been merged into this branch.
So I resolve this conversation.